v2/internal/reconcilers/generic/generic_reconciler.go

```diff
@@ -234,8 +235,32 @@ func (gr *GenericReconciler) createOrUpdate(ctx context.Context, log logr.Logger
 		return ctrl.Result{}, err
 	}
 
-	// Check the reconcile-policy to ensure we're allowed to issue a CreateOrUpdate
-	reconcilePolicy := reconcilers.GetReconcilePolicy(metaObj, log, gr.Config.DefaultReconcilePolicy)
+	var reconcilePolicy annotations.ReconcilePolicyValue
+
+	if gr.Config.ReadReconciliationPolicyFromNamespace {
+		log.V(Debug).Info("Looking for namespace annotations")
+		// Claim resource namespace
+		namespaceObject, err := gr.KubeClient.GetObject(ctx, types.NamespacedName{Namespace: metaObj.GetNamespace(), Name: metaObj.GetNamespace()}, schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Namespace"})
```

Comment:
Doing this launches a new informer for Namespaces, because we don't set `cacheOpts.ReaderFailOnMissingInformer`. I think running a namespace informer here is probably correct... but wondering if, since we're anyway paying the cost of running the namespace informer, we might as well get reconciliation triggered if namespaces are updated.

Doing this correctly requires that we set up a Watch on each resource w/ type namespace, and using the already existing `reconcilers.ARMReconcilerAnnotationChangedPredicate()`, as well as a `handler.EnqueueRequestsFromMapFunc` handler that lists the corresponding resources (in memory from cache) and triggers a reconcile on all of them.

I don't think doing this is all that difficult with the infrastructure we have set up today.